File: solutions/9095.cpp
#include <iostream>

using namespace std;

int t, n, ans;

void count(int num) {
    if (num == 0)
        ans++;
    else {
        if (num >= 1)
            count(num - 1);
        if (num >= 2)
            count(num - 2);
        if (num >= 3)
            count(num - 3);
    }
}

int main() {
    cin >> t;
    for (int i = 0; i < t; i++) {
        cin >> n;
        ans = 0;
        count(n);
        cout << ans << endl;
    }

    return 0;
}

File: solutions/9095.py
import sys

total = 0;


def count(num):
    global total
    
    if num == 0:
        total += 1
        return
    else:
        if num >= 1:
            count(num - 1)
        if num >= 2:
            count(num - 2)
        if num >= 3:
            count(num - 3)


def main():
    global total
    n = int(sys.stdin.readline())

    for _ in range(n):
        num = int(sys.stdin.readline())
        total = 0
        
        count(num)
        print(total)


if __name__=="__main__":
    main()
